### DISCUSSION TITLE
Mesh with Fasets

### DISCUSSION THREAD

**Discussion by ThomasChauve** [neperfepx/neper#474] [Q&A] [answered]
Hello,
I try to create a mesh using neper 4.4.2-16 from git.

```
neper -T -n 10 -dim 2 -id 1 -morpho gg -format vtk,tess -domain "square(1,2)" -tesrsize 128 
neper -M n10-id1.tess  -dim 3 -order 1  -cl "0.02" -faset "faces"
 ```
It works fine but I do not have x0,x1,y0,y1,z0,z1 `$Fasets` in the mesh file. And I like to have those. Is there an option to this.

Best
Thomas

**Comment by rquey** [comment, marked as answer]
Hi Thomas,

`-faset` does not work for 3D mesh obtained by extrusion (i.e. from a 2D tess), because Neper does not know the 3D topology, but you can proceed as follows:
```
neper -T -n 10 -dim 2 -id 1 -morpho gg -format vtk,tess -domain "square(1,2)" -tesrsize 128
neper -T -n 10 -id 1 -morphooptiini "file(n10-id1.tess)" -format vtk,tess -domain "cube(1,2,0.02)"
neper -M n10-id1.tess -order 1 -cl "0.02" -faset "faces"
```

**Reply by ThomasChauve** [reply]
Hi Romain
Thank, it works like a charm.
